protocol/messenger_communities.go

```diff
@@ -2043,6 +2044,57 @@ func (m *Messenger) SetCommunityShard(request *requests.SetCommunityShard) (*Mes
 	return response, nil
 }
 
+func (m *Messenger) SetCommunityMailServers(request *requests.SetCommunityMailServers) (*MessengerResponse, error) {
+	if err := request.Validate(); err != nil {
+		return nil, err
+	}
+	community, err := m.communitiesManager.GetByID(request.CommunityID)
+	if err != nil {
+		return nil, err
+	}
+	if community == nil {
+		return nil, communities.ErrOrgNotFound
+	}
+	if !community.IsControlNode() {
+		return nil, errors.New("not admin or owner")
+	}
+
+	err = m.mailserversDatabase.SaveMailserversForCommunity(request.CommunityID, request.MailServers)
+	if err != nil {
+		return nil, err
+	}
+	if err := m.communityStorenodes.ReloadFromDB(); err != nil {
+		return nil, err
+	}
+	response := &MessengerResponse{
+		Mailservers: request.MailServers,
+	}
+	return response, nil
+}
+
+func (m *Messenger) GetCommunityMailServers(id types.HexBytes) (*MessengerResponse, error) {
+	community, err := m.communitiesManager.GetByID(id)
+	if err != nil {
+		return nil, err
+	}
+	if community == nil {
+		return nil, communities.ErrOrgNotFound
+	}
+	if !community.IsControlNode() {
+		return nil, errors.New("not admin or owner")
+	}
```

Comment:
All members of a community should be able to retrieve the community mailservers. This condition is probably not necessary